google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java

```diff
@@ -187,11 +187,11 @@ public String getTopicNameString() {
    * @param message the message to publish.
    * @return the message ID wrapped in a future.
    */
-  public ApiFuture<String> publish(PubsubMessage message) {
+  public ApiFuture<String> publish(PubsubMessage originalMessage) {
     if (shutdown.get()) {
       throw new IllegalStateException("Cannot publish on a shut-down publisher.");
     }
-
+    PubsubMessage message = OpenCensusUtil.putOpenCensusAttributes(originalMessage);
```

Comment:
Is there any way to hook into OpenCensus' sampling logic here and avoid increasing request size if the span will not be published?